pallets/runtime/testnet-v1/src/fee_details.rs

```diff
@@ -233,8 +233,10 @@ fn is_auth_valid(
                 if let AuthorizationData::JoinIdentity(did) = auth.authorization_data {
                     // make sure that the auth was created by the master key of an identity with valid CDD
                     let master = Identity::did_records(&did).master_key;
-                    if auth.authorized_by == Signatory::from(master) {
-                        return check_cdd(&did);
+                    let master_signatory = Signatory::AccountKey(master);
+                    if auth.authorized_by == master_signatory {
+                        let _ = check_cdd(&did)?;
```

Comment:
Not essential but this could be `return check_cdd(&did).map(|_| Some(master_signatory))?;`

Reply:
Thanks - this is a useful Rust tip! I won't change it here as we've tested with the current code, and this change will not be pushed back into the develop branch (it is a temporary change for Aldebaran).